ql/pricingengines/vanilla/fdblackscholesshoutengine.cpp

```diff
@@ -40,32 +41,59 @@ namespace QuantLib {
         const FdmSchemeDesc& schemeDesc)
     : process_(std::move(process)), tGrid_(tGrid), xGrid_(xGrid), dampingSteps_(dampingSteps),
       schemeDesc_(schemeDesc) {
-
         registerWith(process_);
     }
 
     void FdBlackScholesShoutEngine::calculate() const {
         const Date exerciseDate = arguments_.exercise->lastDate();
         const Time maturity = process_->time(exerciseDate);
+        const Date settlementDate = process_->riskFreeRate()->referenceDate();
+
+        using namespace ext::placeholders;
+        const auto escrowedDividendAdj =
+            ext::make_shared<EscrowedDividendAdjustment>(
+                arguments_.cashFlow,
+                process_->riskFreeRate(),
+                process_->dividendYield(),
+                ext::bind(&GeneralizedBlackScholesProcess::time, process_, _1),
```

Comment:
Now that lambdas are available, I think
```
                [=](Date d){ return process_->time(d); },
```
might be more readable.  What do you think?

Reply:
Sure, I need to put a sticker on my desk, "QL is now C++11";-)